src/libraries/System.ComponentModel.Primitives/src/System/ComponentModel/DesignerAttribute.cs

```diff
@@ -89,11 +95,14 @@ public DesignerAttribute(Type designerType, Type designerBaseType)
         /// <summary>
         /// Gets the name of the base type of this designer.
         /// </summary>
+        // Using PublicParameterlessConstructor to preserve the type. See https://github.com/mono/linker/issues/1878
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)]
         public string DesignerBaseTypeName { get; }
 
         /// <summary>
         /// Gets the name of the designer type associated with this designer attribute.
         /// </summary>
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicParameterlessConstructor)]
```

Comment:
I wonder if we want this actually. Are the designers ever used at runtime?
The problem is that this will basically force include the designer types and some code from them into apps - but they will never be used (since they're designers, those should not run at runtime).
Also - the only method which accesses this TypeDescriptor.CreateDesigner is already marked with RUC - and I think we should not allow calling it on a trimmed app.

Technically the annotation is correct, I'm questioning if it's the right thing to do for size.

Reply:
> the only method which accesses this TypeDescriptor.CreateDesigner is already marked with RUC - and I think we should not allow calling it on a trimmed app.

Just because a method is marked with RUC doesn't mean it can't be called in a trimmed app.

Maybe if this becomes a size issue for WinForms, they can strip out `DesignerAttribute` instances by default - similar to how we strip out some attributes in Blazor WASM that aren't needed at runtime.

Reply:
It's a good point that a better size solution is to strip out the attribute as a whole.

Reply:
These attributes can be and are used at runtime by [`PropertyGrid` control](https://docs.microsoft.com/dotnet/api/system.windows.forms.propertygrid), and can be used by custom designers (e.g. a report viewer)